File: Attendr-main/frontend/src/data/helpArticles.ts
export interface HelpArticle {
  slug:    string;
  title:   string;
  icon:    string;
  steps:   string[];
  related: string[];
}

export const helpArticles: HelpArticle[] = [
  {
    slug:  "otp-not-received",
    title: "Why didn't I receive an OTP?",
    icon:  "MessageSquare",
    steps: [
      "Check that you entered the correct 10-digit mobile number starting with 6, 7, 8, or 9.",
      "Ensure your phone has network signal and is not in airplane mode or DND.",
      "Wait up to 2 minutes — SMS delivery can sometimes be delayed due to network congestion.",
      "Check your spam or blocked messages folder — some carriers block promotional SMS.",
      "If you still haven't received it, tap 'Resend OTP' after the 60-second timer expires.",
      "If the problem persists, contact your HR administrator or reach out to Attendr support.",
    ],
    related: ["invalid-team-id", "location-permission"],
  },
  {
    slug:  "invalid-team-id",
    title: "Invalid Team ID / Organization Not Found",
    icon:  "Building2",
    steps: [
      "Your Team ID is a unique code like 'ABC001' given to you by your HR or manager.",
      "Make sure you're entering the exact Team ID — it is case-insensitive but must match exactly.",
      "Do not add spaces before or after the Team ID.",
      "If you don't know your Team ID, ask your HR manager or company administrator.",
      "Your administrator can find the Team ID in the Attendr Admin panel under Company Settings.",
      "If your HR just registered the company, make sure they have completed the registration.",
    ],
    related: ["unable-to-join-org"],
  },
  {
    slug:  "unable-to-join-org",
    title: "Unable to Join Organization",
    icon:  "Users",
    steps: [
      "Ensure your mobile number has been added to the company by your HR/admin.",
      "Ask your HR to add you in the Employees section of the Admin panel.",
      "Check that your account has not been deactivated — contact HR if you see a deactivation message.",
      "Make sure you are using the same mobile number that was registered with HR.",
      "Try clearing your browser cache and reloading the app.",
      "If nothing works, contact Attendr support with your Team ID and mobile number.",
    ],
    related: ["invalid-team-id", "otp-not-received"],
  },
  {
    slug:  "changed-phone",
    title: "Changed Phone / New Device",
    icon:  "Smartphone",
    steps: [
      "Good news — Attendr works on any device! There is no device lock for employees.",
      "Simply open Attendr on your new phone and go through the normal registration process.",
      "Enter your Full Name, Mobile Number, and Team ID as usual.",
      "Request and verify your OTP — this will log you in on the new device automatically.",
      "Your attendance history will remain intact as it is stored in the cloud.",
      "If you face any issues, contact your HR manager.",
    ],
    related: ["otp-not-received"],
  },
  {
    slug:  "location-permission",
    title: "Location Permission Required",
    icon:  "MapPin",
    steps: [
      "Attendr needs your location to verify that you are within the office premises before marking attendance.",
      "On Android: Go to Settings → Apps → Browser (Chrome/Firefox) → Permissions → Location → Allow.",
      "On iOS/iPhone: Go to Settings → Privacy → Location Services → Safari → While Using the App.",
      "When you open Attendr and tap 'Check In', your browser will ask for location permission — tap Allow.",
      "If you accidentally denied permission, you must reset it in your phone settings (see steps above).",
      "Make sure you are physically present at the office when checking in — GPS must detect you within the defined area.",
    ],
    related: ["app-not-working"],
  },
  {
    slug:  "app-not-working",
    title: "App Not Working Properly",
    icon:  "RefreshCw",
    steps: [
      "First, check your internet connection — Attendr requires an active internet connection.",
      "Try refreshing the page by pulling down (mobile) or pressing F5 (desktop).",
      "Clear your browser cache: Settings → Privacy → Clear browsing data → Cached images/files.",
      "Try opening Attendr in a different browser (Chrome is recommended for best experience).",
      "If you see a blank screen, try force-closing the browser and reopening.",
      "If the problem persists, contact Attendr support with a description of the issue and a screenshot.",
    ],
    related: ["otp-not-received", "location-permission"],
  },
];

export function getArticle(slug: string): HelpArticle | undefined {
  return helpArticles.find((a) => a.slug === slug);
}

export function getRelatedArticles(slugs: string[]): HelpArticle[] {
  return slugs.map((s) => getArticle(s)).filter(Boolean) as HelpArticle[];
}

File: Attendr-main/app/src/main/java/com/trybild/attendr/ui/help/HelpArticles.kt
package com.trybild.attendr.ui.help

data class HelpSection(val heading: String, val bullets: List<String>)
data class HelpArticle(
    val slug: String,
    val title: String,
    val sections: List<HelpSection>,
    val relatedSlugs: List<String>
)

object HelpArticles {
    val all = listOf(
        HelpArticle(
            slug = "otp-not-received",
            title = "OTP Not Received",
            sections = listOf(
                HelpSection("Common Reasons", listOf(
                    "Poor network signal at your location",
                    "SMS delivery delay from your carrier",
                    "Phone number entered incorrectly",
                    "SMS blocked by device DND settings"
                )),
                HelpSection("Steps to Fix", listOf(
                    "Wait 60 seconds before requesting a resend",
                    "Check your network connection and try again",
                    "Verify the phone number is exactly 10 digits",
                    "Disable Do Not Disturb mode and retry",
                    "Check if your inbox is full"
                )),
                HelpSection("Still Not Working?", listOf(
                    "Contact your organization administrator",
                    "Use the Contact Support form below"
                ))
            ),
            relatedSlugs = listOf("invalid-team-id", "app-not-working")
        ),
        HelpArticle(
            slug = "invalid-team-id",
            title = "Invalid Team ID",
            sections = listOf(
                HelpSection("What is a Team ID?", listOf(
                    "Your Team ID (also called Organization ID) is provided by your employer",
                    "It is case-sensitive — enter it exactly as given",
                    "Examples: ABC001, TRYBILD101, ATTENDR500"
                )),
                HelpSection("How to Fix", listOf(
                    "Double-check the ID with your HR or manager",
                    "Make sure there are no extra spaces before or after",
                    "Try uppercase letters if lowercase does not work"
                ))
            ),
            relatedSlugs = listOf("unable-to-join-org", "otp-not-received")
        ),
        HelpArticle(
            slug = "unable-to-join-org",
            title = "Unable to Join Organization",
            sections = listOf(
                HelpSection("Possible Causes", listOf(
                    "Your phone number has not been added to the organization by admin",
                    "The organization account may be inactive or suspended",
                    "You may be using the wrong Team ID"
                )),
                HelpSection("What to Do", listOf(
                    "Ask your administrator to verify your phone number is registered",
                    "Ask admin to confirm the correct Team ID",
                    "If recently onboarded, wait 10 minutes and try again"
                ))
            ),
            relatedSlugs = listOf("invalid-team-id", "changed-phone")
        ),
        HelpArticle(
            slug = "changed-phone",
            title = "Changed Phone / New Device",
            sections = listOf(
                HelpSection("What Happens When You Change Phones", listOf(
                    "Your old session is automatically invalidated for security",
                    "You need to re-register on the new device",
                    "Your attendance history is preserved"
                )),
                HelpSection("Steps to Re-Register", listOf(
                    "Open Attendr on your new device",
                    "Enter your registered phone number",
                    "Enter the OTP received on your number",
                    "Enter your Organization ID and continue"
                ))
            ),
            relatedSlugs = listOf("otp-not-received", "unable-to-join-org")
        ),
        HelpArticle(
            slug = "location-permission",
            title = "Location Permission Required",
            sections = listOf(
                HelpSection("Why Location is Needed", listOf(
                    "Attendr uses GPS to verify you are at the correct work location",
                    "Location is only captured at the moment of check-in or check-out",
                    "No background location tracking is performed"
                )),
                HelpSection("How to Enable Location", listOf(
                    "Go to your phone Settings → Apps → Attendr → Permissions",
                    "Enable Location permission",
                    "Return to Attendr and try checking in again"
                )),
                HelpSection("Mock Location Detected", listOf(
                    "Attendr detects and flags mock/fake GPS apps",
                    "Disable any GPS spoofing or mock location apps",
                    "Turn off Developer Options mock location setting"
                ))
            ),
            relatedSlugs = listOf("app-not-working")
        ),
        HelpArticle(
            slug = "app-not-working",
            title = "App Not Working Properly",
            sections = listOf(
                HelpSection("Quick Fixes", listOf(
                    "Close the app completely and reopen it",
                    "Check your internet connection",
                    "Ensure your phone's date and time are set to automatic"
                )),
                HelpSection("If the Problem Persists", listOf(
                    "Clear the app's cache from Settings → Apps → Attendr → Storage",
                    "Uninstall and reinstall the app from the Play Store",
                    "Make sure your Android version is 7.0 or above"
                ))
            ),
            relatedSlugs = listOf("location-permission", "otp-not-received")
        )
    )

    val bySlug: Map<String, HelpArticle> = all.associateBy { it.slug }

    val listItems = listOf(
        HelpListItem("otp-not-received", "OTP Not Received", "lock"),
        HelpListItem("invalid-team-id", "Invalid Team ID", "group"),
        HelpListItem("unable-to-join-org", "Unable to Join Organization", "business"),
        HelpListItem("changed-phone", "Changed Phone / New Device", "smartphone"),
        HelpListItem("location-permission", "Location Permission Required", "location_on"),
        HelpListItem("app-not-working", "App Not Working Properly", "bug_report")
    )
}

data class HelpListItem(val slug: String, val title: String, val iconKey: String)
